### PR TITLE
Runtime crd

### PR DESCRIPTION
**Pull request by yurado**

### CODE REVIEW

File: api/v1/common_specs/events_gateway_types.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package common_specs
```
Comment (yurado) [inline api/v1/common_specs/events_gateway_types.go:1]: rename file to more appropriate name

File: api/v1/cbcontainersruntime_types.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	Probes common_specs.CBContainersHTTPProbesSpec `json:"probes,omitempty"`
+	// +kubebuilder:default:=<>
+	Prometheus common_specs.CBContainersPrometheusSpec `json:"prometheus,omitempty"`
+}
+
+type CBContainersRuntimeSensorSpec struct {
+	// +kubebuilder:default:=<>
+	Labels map[string]string `json:"labels,omitempty"`
+	// +kubebuilder:default:=<>
+	DeploymentAnnotations map[string]string `json:"deploymentAnnotations,omitempty"`
```
Comment (yurado) [inline api/v1/cbcontainersruntime_types.go:51]: change to daemonSetAnnotations

File: api/v1/cbcontainersruntime_types.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+type CBContainersRuntimeSpec struct {
+	Version string `json:"version,required"`
+	// +kubebuilder:default:="cbcontainers-access-token"
+	AccessTokenSecretName string `json:"accessTokenSecretName,omitempty"`
+	// +kubebuilder:default:=<>
+	ResolverSpec CBContainersRuntimeResolverSpec `json:"controllerSpec,omitempty"`
+	// +kubebuilder:default:=<>
+	SensorSpec CBContainersRuntimeSensorSpec `json:"workerSpec,omitempty"`
+	// +kubebuilder:default:=443
+	InternalGrpcPort int32 `json:"internalGrpcPort,required"`
```
Comment (yurado) [inline api/v1/cbcontainersruntime_types.go:78]: change to omit empty

File: cbcontainers/state/common/objects.go [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 	"reflect"
 	"strconv"
+
+	"github.com/vmware/cbcontainers-operator/api/v1/common_specs"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func mutateEnvVars(container *coreV1.Container, desiredEnvsValues map[string]string, accessTokenSecretName string, eventsGatewaySpec *cbcontainersv1.CBContainersHardeningEventsGatewaySpec, customEnvsToAdd ...coreV1.EnvVar) {
+func MutateEnvVars(container *coreV1.Container, desiredEnvsValues map[string]string, accessTokenSecretName string, eventsGatewaySpec *common_specs.CBContainersEventsGatewaySpec, customEnvsToAdd ...coreV1.EnvVar) {
```
Comment (yurado) [inline cbcontainers/state/common/objects.go:13]: consider changing that to a builder

File: cbcontainers/state/runtime/objects/sensor_daemon_set.go [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	daemonSet.ObjectMeta.Labels = desiredLabels
+	daemonSet.Spec.Selector.MatchLabels = desiredLabels
+	daemonSet.Spec.Template.ObjectMeta.Labels = desiredLabels
+	daemonSet.Spec.Template.Spec.ServiceAccountName = commonState.DataPlaneServiceAccountName
+	daemonSet.Spec.Template.Spec.PriorityClassName = commonState.DataPlanePriorityClassName
+	daemonSet.Spec.Template.Spec.ImagePullSecrets = []coreV1.LocalObjectReference{{Name: commonState.RegistrySecretName}}
+
+	daemonSet.Spec.Template.Spec.DNSPolicy = coreV1.DNSClusterFirstWithHostNet
```
Comment (yurado) [inline cbcontainers/state/runtime/objects/sensor_daemon_set.go:79]: move to const

File: cbcontainers/state/runtime/objects/sensor_daemon_set.go [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+	container.Args = []string{sensorVerbosityFlag, fmt.Sprintf("%d", sensorSpec.VerbosityLevel)}
+	container.Command = []string{sensorRunCommand}
+	commonState.MutateImage(container, sensorSpec.Image, version)
+	commonState.MutateContainerFileProbes(container, sensorSpec.Probes)
+	obj.mutateEnvVars(container, sensorSpec, accessTokenSecretName, desiredGRPCPortValue)
+	obj.mutateContainerPorts(container, desiredGRPCPortValue)
+	obj.mutateSecurityContext(container)
+}
+
+func (obj *SensorDaemonSetK8sObject) mutateContainerPorts(container *coreV1.Container, desiredGRPCPortValue int32) {
```
Comment (yurado) [inline cbcontainers/state/runtime/objects/sensor_daemon_set.go:145]: remove

File: cbcontainers/state/runtime/runtime_state_applier.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	return &RuntimeStateApplier{
+		resolverDeployment: runtimeObjects.NewResolverDeploymentK8sObject(),
+		resolverService:    runtimeObjects.NewResolverServiceK8sObject(),
+		sensorDaemonSet:    runtimeObjects.NewSensorDaemonSetK8sObject(),
+		childApplier:       childApplier,
+		log:                log,
+	}
+}
+
+func (c *RuntimeStateApplier) ApplyDesiredState(ctx context.Context, cbContainersRuntime *cbcontainersv1.CBContainersRuntime, client client.Client, setOwner applymentOptions.OwnerSetter) (bool, error) {
```
Comment (yurado) [inline cbcontainers/state/runtime/runtime_state_applier.go:37]: add applySensor

**Review by BenRub — APPROVED** [review]

**Pull request merged**